Create en 2020report (#7)
vivliostyle/vivliostyle_doc: en/reports/vivliostyle-report-2020/vf2020report.md

---
title: "Vivliostyle Foundation FY2020 Activity Report"
lang: en
---

## **Chapter 1: Activity Report for FY2020**

(The 3rd Fiscal Year: from April 1, 2020 to March 31, 2021)

### 1. Organize developer/user events

We have decided to hold developer/user events twice a year as a place to widely let users and developers know about Vivliostyle. This year, we were able to hold the following events.

- [April 4, 2020: Vivliostyle User/Dev Meetup 2020 Spring (Held online)](https://vivliostyle.connpass.com/event/170939/)
- [October 24, 2020: Vivliostyle User/Dev Meetup 2020 Autumn (Held online)](https://vivliostyle.connpass.com/event/189940/)

> The spring event coincided with the COVID 19 calamity. Initially, the event was scheduled to be held at the Japan Association of Graphic Arts Technology (JAGAT), but as Avoid the “Three Cs” was becoming a top priority, we were able to get through it by switching to online distribution. While most of the similar events were cancelled, thankfully, 142 people registered for the event. In terms of content, we were able to inform people about the real activities of our corporation, focusing on the report of Vivliostyle Pub, which was in the midst of alpha development at the time.

Reviewer: online distribution? Do you mean you held the event online?

"calamity" is an odd choice here. Maybe "disaster" or just "pandemic"?

> On the other hand, the fall event was a thoughtful seminar on CSS typesetting. For example, [Simon Worthington](https://www.force11.org/users/simon-worthington)of the Open Science Lab - German National Library of Science and Technology reported on Rapid publishing for public health books against COVID 19. And our contributor [akabeko](https://github.com/akabekobeko) reported on Vivliostyle's competitor [Paged.js](https://www.pagedjs.org/). In this way, we were able to show the appeal of CSS typesetting beyond Vivliostyle.

Reviewer: What do you mean by "thoughtful"?

Author: I see, it is indeed a difficult expression to understand. How about the following?

On the other hand, the autumn event seemed to be a good opportunity to work on CSS typesetting as a seminar on the subject.

Reviewer: Sounds good

lang: en
---

## **Chapter 1: Activity Report for FY2020**

(The 3rd Fiscal Year: from April 1, 2020 to March 31, 2021)

### 1. Organize developer/user events

We have decided to hold developer/user events twice a year as a place to widely let users and developers know about Vivliostyle. This year, we were able to hold the following events.

- [April 4, 2020: Vivliostyle User/Dev Meetup 2020 Spring (Held online)](https://vivliostyle.connpass.com/event/170939/)
- [October 24, 2020: Vivliostyle User/Dev Meetup 2020 Autumn (Held online)](https://vivliostyle.connpass.com/event/189940/)

The spring event coincided with the COVID 19 calamity. Initially, the event was scheduled to be held at the Japan Association of Graphic Arts Technology (JAGAT), but as Avoid the “Three Cs” was becoming a top priority, we were able to get through it by switching to online distribution. While most of the similar events were cancelled, thankfully, 142 people registered for the event. In terms of content, we were able to inform people about the real activities of our corporation, focusing on the report of Vivliostyle Pub, which was in the midst of alpha development at the time.

> On the other hand, the fall event was a thoughtful seminar on CSS typesetting. For example, [Simon Worthington](https://www.force11.org/users/simon-worthington)of the Open Science Lab - German National Library of Science and Technology reported on Rapid publishing for public health books against COVID 19. And our contributor [akabeko](https://github.com/akabekobeko) reported on Vivliostyle's competitor [Paged.js](https://www.pagedmedia.org/paged-js/). In this way, we were able to show the appeal of CSS typesetting beyond Vivliostyle.

Reviewer 2: Paged.js のURL 
https://www.pagedmedia.org/paged-js/
は "Deprecating" と出るように古いURLです。新しいURLは
https://www.pagedjs.org/

```suggestion
On the other hand, the fall event was a thoughtful seminar on CSS typesetting. For example, [Simon Worthington](https://www.force11.org/users/simon-worthington)of the Open Science Lab - German National Library of Science and Technology reported on Rapid publishing for public health books against COVID 19. And our contributor [akabeko](https://github.com/akabekobeko) reported on Vivliostyle's competitor [Paged.js](https://www.pagedjs.org/). In this way, we were able to show the appeal of CSS typesetting beyond Vivliostyle.
```

### 2. Official website renewal

> As briefly mentioned in [the previous year's activity report](https://vivliostyle.github.io/vivliostyle_doc/en/reports/vivliostyle-report-2019/vf2019report.html), the design of the official website had remained unchanged for over two years since the founding of the corporation. In order to revamp it, we had a series of discussions at the developper meeting and decided on the following policy, and commissioned [yamasy1549](https://github.com/yamasy1549) to start work on it in December 2019.

Reviewer 2: typo: developper → developer

会議は１回だけでないので、
meeting → meetings
でしょう

```suggestion
As briefly mentioned in [the previous year's activity report](https://vivliostyle.github.io/vivliostyle_doc/en/reports/vivliostyle-report-2019/vf2019report.html), the design of the official website had remained unchanged for over two years since the founding of the corporation. In order to revamp it, we had a series of discussions at the developer meetings and decided on the following policy, and commissioned [yamasy1549](https://github.com/yamasy1549) to start work on it in December 2019.
```

- Single column layout for mobile viewing.
- The top page of the site features an animation that provides an intuitive understanding of what Vivliostyle is all about.
- Place the GitHub repository link prominently on each page.
- For the convenience of users, the following pages will be newly added.
    - “Getting Started” that summarizes the product outline. 
    - Detailed development documentation, including “User Guides” and “Tutorial Guides”.
    - “FAQ” to solve users' doubts.

> It was difficult to prepare for the developer/user event (see the previous section) at the same time, but we managed to reopen the site on April 3, the day before the event. After that, the following information was added.

Reviewer 2: "reopen" ではそれまでサイトが閉鎖していて再開したかのようです。"renew" がよいかと思います。

```suggestion
It was difficult to prepare for the developer/user event (see the previous section) at the same time, but we managed to renew the site on April 3, the day before the event. After that, the following information was added.
```

> -2020/4/10: [Notified that the German National Library of Science and Technology has published a textbook on COVID-19 measures using Vivliostyle.](https://vivliostyle.org/blog/2020/04/10/tib-book-against-covid19/)

Reviewer 2: 箇条書きの "-" のあとにスペースがないのはエラーです。

それから、ここに限りませんが、日付のフォーマットは ISO 8601 に従う "2020-04-10" の形式に統一するのがよいかと思います。

参考: 
https://en.wikipedia.org/wiki/Date_format_by_country
> The ISO 8601 format YYYY-MM-DD (2021-06-23) is intended to harmonize these formats and ensure accuracy in all situations. Many countries have adopted it as their sole official date format, though even in these areas writers may adopt abbreviated formats that are no longer recommended.

Author: > ここに限りませんが、日付のフォーマットは ISO 8601 に従う "2020-04-10" の形式に統一するのがよいかと思います。

じつは最初 YYYY-MM-DD で書き進めていたのですが、以下の部分で困ってしまいました。

> - [Vivliostyle CLI v3.0 new spec (2020/11/9-2021/2/9)](https://github.com/vivliostyle/community/wiki/Vivliostyle-CLI-v3.0-new-spec)

範囲を示す「-」が、年月日の区切りと重複してしまいます。
そこでブログなどで使われている表記方法、YYYY/MM/DD  を使うことにしました。
こういう場合、 ISO 8601ではどう表記するのでしょうか？

Reviewer 2: https://stackoverflow.com/questions/20413843/is-there-any-kind-of-standard-for-representing-date-ranges に同様の質問と答えがあります。

ISO 8601 では期間は 2020-11-09/2021-02-09 のようにスラッシュを使うことになっています。 https://ja.wikipedia.org/wiki/ISO_8601#%E6%9C%9F%E9%96%93

また、 スラッシュ "/" の代用としてハイフン２つ "--" でも可とのことです。
https://en.wikipedia.org/wiki/ISO_8601#Time_intervals
>  "A solidus may be replaced by a double hyphen ["--"] by mutual agreement of the communicating partners."

ハイフン２つで
2020-11-09--2021-02-09
のように書いた方が分かりやすい気がします。

Author: ありがとうございます！　表記方法があったのですね。
そのように修正します。

Reviewer 2: "2020-4-10" は ISO 8601 形式として正しくないです。"2020-04-10" のように、月と日は必ず2桁にする必要があります。

- 2020/4/29: [Called for support from GitHub Sponsors.](https://vivliostyle.org/blog/2020/04/29/become-a-sponsor-to-vivliostyle-via-github-sponsors/)
- 2020/6/11: [2019 Activity Report released.](https://vivliostyle.github.io/vivliostyle_doc/en/reports/vivliostyle-report-2019/vf2019report.html)

> - 2020/8/19: [Creator Book Tutorial Guide Released.](https://vivliostyle.org/make-books-with-create-book/)

Reviewer 2: Creator Book → Create Book

- 2020/8/29: [Announced the start of accepting donations in Japanese yen by credit card.](https://vivliostyle.org/blog/2020/11/20/You-can-support-Vivliostyle-without-a-GitHub-account/)

In particular, the appeal for donations is important for a general incorporated association that does not make a profit (and does not distribute profits). In fact, we were able to collect 61,209 yen in total this fiscal year (Fig. 1), which is about 4% of our total ordinary income. This is still not a large amount, but we would like to continue to nurture it as a receptacle for people who support our activities.

![Figure 1: Breakdown of ordinary income for the current fiscal year](./img/fig-1.png){ width=50% }
 
### 3. Product Development Status

In this section, we will explain the development status of each product for this fiscal year.

#### [Vivliostyle Core / Viewer (vivliostyle.js)](https://github.com/vivliostyle/vivliostyle.js)

> We were able to add many functions to Vivliostyle Core and Vivliostyle Viewer (Vivliostyle.js), which are the foundation of the entire product, this year. As a result, we were able to promote support for [CSS Paged Media](https://www.w3.org/TR/css-page-3/), which had not been sufficiently supported in the past. The main features we added are as follows.

Reviewer 2: プロダクトは１つだけではないので複数形にするべきでしょう：
the entire product → the entire products

 - Implementation of a slide bar for page navigation, and support for mouse wheel. ([v2.2.0 / 2020/11/26](https://github.com/vivliostyle/vivliostyle.js/releases/tag/v2.2.0))
- Support for specifying EPUB spread view. ([v2.3.0 / 2020/12/7](https://github.com/vivliostyle/vivliostyle.js/releases/tag/v2.3.0))
- Support for named strings to realize headers, footers, etc. ([v2.4.0 / 2020/12/28](https://github.com/vivliostyle/vivliostyle.js/releases/tag/v2.4.0))
- Support for n-th page selector. ([v2.5.0 / 2021/2/26](https://github.com/vivliostyle/vivliostyle.js/releases/tag/v2.5.0))

> - 印刷ボタンの追加 ([v2.6.0 / 2021/3/14](https://github.com/vivliostyle/vivliostyle.js/releases/tag/v2.6.0))

Reviewer 2: 印刷ボタンの追加 → Added the Print button

#### [Vivliostyle CLI](https://github.com/vivliostyle/vivliostyle-cli)

> This is a converter that allows CSS typesetting with a command line interface. With the addition of support for Markdown format input this fiscal year, the scope of its use has expanded dramatically. In addition, since the preview screen has been integrated into Vivliostyle Viewer, it can be said that Vivliostyle Viewer has grown into a product that allows users to read long texts, not just to check the formatting results. As will be discussed later, this is expected to have a positive impact on the future Vivliostyle Pub v1. The details for each release are as follows.

Reviewer 2: > In addition, since the preview screen has been integrated into Vivliostyle Viewer, it can be said that Vivliostyle Viewer has grown into a product that allows users to read long texts, not just to check the formatting results.

この英文では、「Vivliostyle Viewerは、単に組版結果を確認するだけでなく、長文を読むことができる製品に成長した」という意味になるが、主語が間違っているのでないか？　ここでの「製品」はVivliostyle ViewerではなくてVivliostyle CLIのはずです。

元の日本語は、
> また、プレビュー画面がVivliostyle Viewerに統合されたことで、単に組版結果を確認できるだけでなく「長文を読める」プロダクトに成長したと言える。

でした。

英文の修正案：
> In addition, with the integration of Vivliostyle Viewer for the preview mode, it has grown into a product that that allows users to read long texts, not just to check the formatting results.

- Replaced the preview UI with Vivliostyle Viewer. ([v3.2.0 / 2021/3/29](https://github.com/vivliostyle/vivliostyle-cli/releases/tag/v3.2.0))

> - Supports input of various file formats including Markdown/ Support for [Themes](https://github.com/vivliostyle/themes)/ Supports multiple outputs, including web books/ Support loading external web pages by URL. ([v3.0.0 / 2021/2/7](https://github.com/vivliostyle/vivliostyle-cli/releases/tag/v3.0.0))

Reviewer 2: "… including Markdown/ Support for …"
のようにスラッシュが前の語についていると、スラッシュで区切っているというのが分かりにくいです。スラッシュを使うなら
"… including Markdown / Support for …"
のようにスラッシュの前後ともにスペースを入れるべきだろうと思います。

スラッシュではなくてセミコロンを使うなら後にだけスペースを入れる
"… including Markdown; Support for …"
でよいでしょう。

#### [Vivliostyle Flavored Markdown (VFM)](https://github.com/vivliostyle/vfm)

This is a Markdown dialect for book typesetting that allows for ruby, image size specification, captions, footnotes, etc. It is upwardly compatible with the most popular Markdown specification, GitHub Flavored Markdown (GFM), and was developed for input data in Vivliostyle CLI v3.0. v1.0.0-alpha.0 was released on June 13, and v1.0.0-alpha.17 was released as of the end of this fiscal year. It has been under continuous development since then.

#### [Themes](https://github.com/vivliostyle/themes)

This is a mechanism to publish/reuse style sheets for CSS typesetting in Vivliostyle CLI as a package. The first official theme was released on July 1, and as of the end of this fiscal year, the following three themes have been released. We will continue to expand them.

- [theme-slide (Slide)](https://github.com/vivliostyle/themes/tree/master/packages/@vivliostyle/theme-slide)

> - [theme-bunko (Vertical Book)](https://github.com/vivliostyle/themes/tree/master/packages/@vivliostyle/theme-bunko)

Reviewer 2: Vertical Book（縦型の本？）では意味が通じないと思います。

Vertical Book → Japanese book with vertical text
でどうでしょう。

As of the end of the previous fiscal year, the biggest concern was securing development funding, and the solution was discussed at the developer meeting. Therefore, the contributors suggested that they apply for the [Mitou Advanced Project](https://www.ipa.go.jp/jinzai/advanced/2020/koubo_index.html) with the theme of developing Vivliostyle Pub. 

As a result of discussions among the members, it was decided to take on the challenge. Since organizations are not allowed to apply due to regulations, we completed the application on April 2 as a group of individuals consisting of [youchan](https://github.com/youchan) and [takanakahiko](https://github.com/takanakahiko), in addition to the aforementioned contributors and Representative Director Murakami.

> From that point on, all of the members of the team worked together, and day after day, night after night, the development of Vivliostyle Pub began. The goal is the second round of review scheduled for mid-May. The following is an overview of the version (alpha version) at this point.

Reviewer 2: "mid-May" というのがいつのことか分かりやすいように "mid-May 2020" と書いた方がよいと思います。

> From that point on, all of the members of the team worked together, and day after day, night after night, the development of Vivliostyle Pub began. The goal is the second round of review scheduled for mid-May. The following is an overview of the version (alpha version) at this point.

Reviewer 2: > all of the members of the team worked together, and day after day, night after night,

日本語版では「全員が一丸となり、夜を日に継いで」という表現でしたが、この英語訳でさらに大袈裟すぎる感じがします。みんな他の仕事がある中で可能な範囲で協力してくれたのであって、チーム全員が連日連夜この開発ばかりやっていたというわけではなかったと思います。私もメンバーのひとりでしたが私自身は直接開発作業をしてません。

Author: GitHubのコミット履歴をみると、実際に4月18日〜5月16日の期間中、多様なメンバーが多くのコミットをしています。この状態を表現しようというのが当該箇所の意図です。日本語版では文字数を圧縮するため、現在のような表現に落ち着きましたが、英語版ではもっと違う表現はありうると思います。

もし可能なら、村上さんの方から違和感のない表現を提案してもらえませんか。

Reviewer 2: その期間のvivliostyle-pubのContributionsのグラフ: 
https://github.com/vivliostyle/vivliostyle-pub/graphs/contributors?from=2020-04-17&to=2020-05-17&type=c
これを見ると、「多様なメンバーが多くのコミット」というより、少数のメンバーによる多くのコミットだったことがわかります。GitHubコミット履歴にあらわれないメンバーたちの働きもあったかとは思いますが。

> もし可能なら、村上さんの方から違和感のない表現を提案してもらえませんか。

日本語：「そこからVivliostyle Pubの集中した開発がはじまった。」
英語：「From that point on, the intensive development work on Vivliostyle Pub began.」

でどうでしょう。

Author: > これを見ると、「多様なメンバーが多くのコミット」というより、少数のメンバーによる多くのコミットだったことがわかります。

なぜ村上さんがこの部分にそれほど拘るのか、訝しんでいます。
たしかに「多様なメンバー」は正確ではなかったかもしれません。それはお詫びします。
しかし開発チームは6名でした。このグラフでは3名が多くのコミットをしていることが分かりますが、6分の3は「少数」なのでしょうか？

> 日本語：「そこからVivliostyle Pubの集中した開発がはじまった。」
> 英語：「From that point on, the intensive development work on Vivliostyle Pub began.」
>
> でどうでしょう。

はい、了解いたしました。

Author: ご指摘に応えて、ja版3-6、Vivliostyle Pubの項を修正しました。`report.css`の第2章合計欄も修正しています。これでOKならen版も修正します。なお、`conv.sh`から出力したHTMLとPDFも追加しています。

Author: 4節の後が改ページになっており、空白が気になります。ここは詰めた方がよいのではないでしょうか？

Reviewer 2: Pubの項の修正内容、了解しました。ありがとうございます。

「理事」で改ページされないようにするにはCSSの次のところ削除するとよいです：

```
#理事 {
  break-before: column;
}
```

```
#directors {
  break-before: column;
}
```

publication.json が追加されてますが、不要なので削除してください。

- The Vivliostyle CLI deployed in the cloud implements a very early parser for VFM.
- Vivliostyle CLI can output the result of CSS typesetting to the browser.

> - Vivliostyle CLI can generate and download a PDF of the CSS typesetting result from the Export menu.

Reviewer 2: 「Vivliostyle CLI はCSS組版の結果をブラウザに出力することができます。」
「Vivliostyle CLIは、ExportメニューからCSS組版結果のPDFを生成してダウンロードすることができます。」
とありますが、どちらもVivliostyle CLIが主語になっているのは間違いです。CLIが組版結果をブラウザに出力するわけではないし、CLIが組版結果をダウンロードするわけでありません。

修正案：

- It can output the result of CSS typesetting to the browser.
- The generated PDF can be downloaded from the Export menu.

![Figure 3 Current Vivliostyle Pub](./img/fig-3.png){ width=50% }

Vivliostyle Pub has almost stopped development, while its components, VFM, Themes, Vivliostyle CLI, and Vivliostyle Core / Viewer, have evolved significantly.  However, in order to do so, it was necessary to overcome the impact of the failure of the Mitou Advanced Project. The following document by Representative Director Murakami served as a roadmap.

- [Vivliostyle Pub v1 Req (2020/11/9)](https://github.com/vivliostyle/community/wiki/Vivliostyle-Pub-v1-Req)
- [Vivliostyle CLI v3.0 new spec (2020/11/9-2021/2/9)](https://github.com/vivliostyle/community/wiki/Vivliostyle-CLI-v3.0-new-spec)

If we can incorporate the VFM, Themes, and Vivliostyle CLI that have been developed in this way, we can successfully release Vivliostyle Pub v1. That will be the task for the next fiscal year. In order to accomplish this, the development is now underway with the aim of releasing the beta version at the end of August 2021 and launching it at the end of December of the same year.

### 4. Collaborate with external organizations

> In the previous fiscal year, the project income was 0 yen. However, as shown in the next chapter and Figure 1 above, we were able to generate 1,503,721 yen in business income this fiscal year. Especially significant was the development contracted by a company that Representative Director Murakami has been involved with for a long time. In fact, many of the development results mentioned in the previous section on Vivliostyle Core / Viewer were paid for by these companies.

Reviewer 2: 複数形に直してください：
contracted by a company → contracted by companies

> In addition, Representative Director Murakami continued to have discussions with Simon Worthington, who was mentioned in Section 1, and continued to commit himself to their development. As a result, he was able to receive development orders from them and earn business income. In this way, collaboration with external organizations led directly to the development of Vivliostyle, and as a result, we were able to proceed with the CSS Paged Media support, which had been an issue. We are very grateful for this.

Reviewer 2: > he was able to receive development orders

個人で受注したわけではないので、
"he was" → "we were"
と直してください。
